### PR TITLE
Elevator puzzle

### LINKED ISSUE #895: Lights Out
Create a game of Lights Out. 
https://en.wikipedia.org/wiki/Lights_Out_(game)

### DoD
The game can be played without issues such as crashing and the game can tell if the player has won.

### PR DESCRIPTION
**Pull request by AncientFungus**
Resolves #895, #896, #897, #906, #907, #909

### CODE REVIEW

**Comment by andens** [comment]
iamheman

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 1]
```diff
@@ -0,0 +1,81 @@
+class elevator_moveCart {
+    Hub @hub;
+    Entity @self;
+	Entity @elevator;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as [hunk 2]
```diff
@@ -0,0 +1,71 @@
+class elevator_frontDoorCon {
+    Hub @hub;
+    Entity @doorWithX;
+	Entity @doorWithService;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as [hunk 3]
```diff
@@ -0,0 +1,57 @@
+class elevator_backDoorCon {
+    Hub @hub;
+    Entity @doorWithX;
+	Entity @doorWithService;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 4]
```diff
@@ -0,0 +1,161 @@
+class LightsOutScript {
+    Entity @board;
+    Entity @rightController;
+	Entity @elevatorCartScript;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/LightsOutScript.as [hunk 5]
```diff
@@ -0,0 +1,161 @@
+class LightsOutScript {
+	Entity @board;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/LightsOutScript.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as [hunk 6]
```diff
@@ -0,0 +1,57 @@
+class elevator_backDoorCon {
+	Hub @hub;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_backDoorCon.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as [hunk 7]
```diff
@@ -0,0 +1,71 @@
+class elevator_frontDoorCon {
+	Hub @hub;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_frontDoorCon.as]: Horizontal tab

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 8]
```diff
@@ -0,0 +1,81 @@
+class elevator_moveCart {
+	Hub @hub;
```
Comment (Chainsawkitten) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Horizontal tab

**Pull request merged**